src/runtime/virtcontainers/utils/utils.go

```diff
@@ -122,13 +122,12 @@ func WriteToFile(path string, data []byte) error {
 }
 
 // CalculateMilliCPUs converts CPU quota and period to milli-CPUs
-func CalculateMilliCPUs(quota int64, period uint64) uint32 {
-
+func CalculateMilliCPUs(quota int64, period uint64) float32 {
 	// If quota is -1, it means the CPU resource request is
 	// unconstrained.  In that case, we don't currently assign
 	// additional CPUs.
 	if quota >= 0 && period != 0 {
-		return uint32((uint64(quota) * 1000) / period)
+		return float32(quota) / float32(period)
```

Comment:
With the 1000 factor removed, it seems that the returned value doesn't match the function's contract.

Reply:
I'm not sure if I follow you here, @gkurz.  Mind to expand?

Reply:
Someone calling `CalculateMilliCPUs()` would legitimately expect to get milli cpus but they would get fractional cpus instead.

Reply:
@fidencio I had the same reaction reading the code, and I don't see a change here. But it's marked as 'resolved'.